pyatlan/model/typedef.py

```diff
@@ -639,6 +657,31 @@ def applicable_glossary_types(self, glossary_types: Set[str]):
             )
         self.options.applicable_glossary_types = json.dumps(list(glossary_types))
 
+    @property
+    def applicable_domain_types(self) -> Set[str]:
+        """
+        Data product type names to which to restrict the attribute.
+        These cover asset types in data products and data domains.
+        Only assets of one of these types will have this attribute available.
+        """
+        if self.options and self.options.applicable_domain_types:
+            return set(json.loads(self.options.applicable_domain_types))
+        return set()
+
+    @applicable_domain_types.setter
+    def applicable_domain_types(self, domain_types: Set[str]):
```

Comment:
I think the type should be Set[Literal["DataDomain", "DataProduct"]] not Set[str]

Reply:
Thanks!